packages/nimble-components/src/anchor-button/tests/anchor-button.stories.ts

```diff
@@ -35,8 +36,8 @@ const metadata: Meta<AnchorButtonArgs> = {
     parameters: {
         docs: {
             description: {
-                component:
-                    'An anchor button is a component with the visual appearance of a button, but it navigates like an anchor/link when pressed.'
+                component: `An anchor button is a component with the visual appearance of a button, but it navigates like an anchor/link when pressed.
+                    ${targetGuidelines}`
             }
```

Comment:
Looking at the button mdx docs I'm also realizing we are accidentally pulling in all mdx files as stories:

![image](https://github.com/ni/nimble/assets/1588923/973987d3-aad5-4a61-8db1-1e67c4e61e00)

Because of:

https://github.com/ni/nimble/blob/fde3626559e2469f8540c45d9ff774838b49bb75/packages/nimble-components/.storybook/main.js#L4-L9

Would be nice in this PR or a tech debt issue + follow-up to rename our mdx files that are actually stories to be `*.stories.mdx` and then update that selector to reflect that:

```js
export const stories = [
    '../docs/**/*.stories.mdx',
    '../docs/**/*.stories.ts',
    '../src/**/*.stories.mdx',
    '../src/**/*.stories.ts'
];
```

Reply:
I made a quick effort to try to rename the relevant files, but I ran into problems. It looks like storybook does not treate `.mdx` files and `.stories.mdx` files the same, so we can't simply change the extension because we want to. That behavior is discussed in this storybook issue: https://github.com/storybookjs/storybook/issues/20496

I can create a tech debt item for fixing the fact that all of our mdx files are showing up in storybook since it looks like we'll need to have a bit more discussion about how we want to distinguish internal/external mdx files.

Reply:
Lovely, thanks for investigating. Yea we can make a tech debt to figure out how to handle those helper mdx files in the storybook nav tree. But I think we should go ahead and use the pattern for the PR even if it's adding a bit more to that tech debt.

Reply:
Created #1419